### ISSUE TITLE
Windows 21.3 does not build with future JDK 17 (17.0.5+4) and JDK 11 (11.0.18+1)

### ISSUE THREAD

**Issue by Karm** [graalvm/mandrel#428]
Both 21.3 JDK17 and 22.3 JDK17 build fine.
```
DEBUG [OperatingSystem] Execute [.\mandrel-java11-21.3.3.0-devde7d96f17e9\lib\svm\bin\native-image.cmd, --macro:native-image-agent-library] in C:\workspace\workspace\mandrel-21-3-jdk11-latest-windows-amd64 with environment variables []
[native-image-agent:5804]    classlist:   2,714.32 ms,  0.96 GB
[native-image-agent:5804]        (cap):   2,598.36 ms,  0.96 GB
The URL protocol jar is not tested and might not work as expected.
Supported URL protocols enabled by default: file,resource. Supported URL protocols available on demand: http,https.
[native-image-agent:5804]        setup:   4,642.20 ms,  0.96 GB
[native-image-agent:5804]     analysis:  10,650.29 ms,  2.33 GB
Fatal error:com.oracle.svm.util.ReflectionUtil$ReflectionUtilError: java.lang.NoSuchMethodException: sun.security.mscapi.CKeyStore.loadKeysOrCertificateChains(java.lang.String)
	at org.graalvm.nativeimage.pointsto/com.oracle.svm.util.ReflectionUtil.lookupMethod(ReflectionUtil.java:70)
	at org.graalvm.nativeimage.pointsto/com.oracle.svm.util.ReflectionUtil.lookupMethod(ReflectionUtil.java:57)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.JNIRegistrationUtil.method(JNIRegistrationUtil.java:93)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.SecurityServicesFeature.registerSunMSCAPIConfig(SecurityServicesFeature.java:416)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ReachabilityHandlerFeature.duringAnalysis(ReachabilityHandlerFeature.java:119)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$12(NativeImageGenerator.java:727)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:73)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:727)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:529)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:488)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:410)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:594)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:124)
Caused by: java.lang.NoSuchMethodException: sun.security.mscapi.CKeyStore.loadKeysOrCertificateChains(java.lang.String)
	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2475)
	at org.graalvm.nativeimage.pointsto/com.oracle.svm.util.ReflectionUtil.lookupMethod(ReflectionUtil.java:62)
	... 12 more
[native-image-agent:5804]      [total]:  18,331.79 ms,  2.33 GB
# Printing build artifacts to: C:\workspace\workspace\mandrel-21-3-jdk11-latest-windows-amd64\mandrel-java11-21.3.3.0-devde7d96f17e9\lib\svm\macros\native-image-agent-library\..\..\..\..\bin\native-image-agent.build_artifacts.txt
Error: Image build request failed with exit status 1
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: Failed, exit code: 1
	at OperatingSystem.exec(build.java:1558)
	at build.buildAgents(build.java:189)
	at build.main(build.java:171)
Caused by: java.lang.RuntimeException: Failed, exit code: 1
	at OperatingSystem.exec(build.java:1543)
	at build.buildAgents(build.java:189)
	at build.main(build.java:171)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.execute(Main.java:404)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.run(Main.java:179)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.main(Main.java:119)
```

@zakkak @jerboaa Is this a concern, or we can abandon this by switching to 22.3.0 JDK17 anyway?

WIP trying 22.3.0 JDK11 windows...

**Comment by jerboaa** [comment]
@Karm This is https://github.com/graalvm/mandrel/issues/419 and needs a backport of the fix in master to 21.3.

**Comment by Karm** [comment]
Thx

**Comment by jerboaa** [comment]
Re-opening. Let's close it once the backport is in.

**Comment by jerboaa** [comment]
I see that https://bugs.openjdk.org/browse/JDK-6782021 got backported to 11.0.18. So it's relevant for the January CPU update of the 21.3 train.

**Comment by jerboaa** [comment]
> I see that https://bugs.openjdk.org/browse/JDK-6782021 got backported to 11.0.18. So it's relevant for the January CPU update of the 21.3 train.

That's not the full story. For JDK 17 based builds we need the fix in `17.0.5` which is October 2022 CPU update.

**Comment by jerboaa** [comment]
Closed with https://github.com/graalvm/mandrel/commit/840012cb045c62b2cd104f9d094a4e0c2c74268e (PR #430).
